views/topics.go

```diff
@@ -99,6 +99,7 @@ func editTopic(w http.ResponseWriter, r *http.Request) {
 		title := r.PostFormValue("title")
 		content := r.PostFormValue("content")
 		isSticky := r.PostFormValue("is_sticky") == "1"
+		isReadOnly := r.PostFormValue("is_readonly") == "1"
```

Comment:
Here, isReadOnly will always be false for regular (non-admin / non-mod) users. So, if a regular user edits a topic, it gets re-opened, which is definitely not what we want!